File: 2026/Árvore de Natal.cpp
#include <bits/stdc++.h>
using namespace std;

void max_self(int& a, int b){
    a = max(a,b);
}
int main (){
    int t;
    cin >> t;
    int k = 1;
    int n, W;
    while (t>0){
        cin >> n >> W;
        vector<int> dp(W+1);
    
        
        for (int item = 0; item < n; item++){
            int value, weight;
            cin >> value >> weight;

            for (int weight_already = W; weight_already>=0; weight_already--){
                
                //cout << "wa " << weight_already <<"w " <<weight <<"wa+ w " << weight_already+weight << endl;
                //cout << "dp wa + value " << dp[weight_already] + value << endl;
                if (weight_already+weight < (W+1))
                    max_self(dp[weight_already+weight],dp[weight_already]+value);
                
            
            }
            
            
        }
       
        int answer = 0;
        for (int i = 0; i <= W; i++){
            max_self(answer, dp[i]);
        }
        cout << "Galho " << k << ":"<<endl;
        cout << "Numero total de enfeites: " << answer << endl;
        cout << endl;
        k++;
        dp.clear();
        t--;
    }
}

File: 2026/Árvore de Natal.py
t = int(input())
k = 1
while t > 0:
    n = int(input())
    W = int(input())

    dp = [0]*(W+1)

    def max_self(a, b):
        return max(a,b)

    for item in range(n):
        value, weight = map(int, input().split())
        for weight_already in range(W, -1, -1):
            if weight_already+weight < (W + 1):
                dp[weight_already+weight] = max_self(dp[weight_already+weight], dp[weight_already]+value)
                #print(dp)
            
        

    answer = 0
    for i in range(W+1):
        answer = max_self(answer, dp[i])
    print("Galho "+str(k)+":")
    print("Numero total de enfeites: " + str(answer))
    print()

    t-=1
    k+=1

